featurebyte/service/offline_store_feature_table_comment.py

```diff
@@ -0,0 +1,172 @@
+class ColumnComment:
+    """
+    Representation of a comment for a specific offline feature table column
+    """
+
+    table_name: str
+    column_name: str
+    comment: str
+
+
+class OfflineStoreFeatureTableCommentService:
+    """
+    OfflineStoreFeatureTableCommentService
+    """
+
+    def __init__(
+        self,
+        entity_service: EntityService,
+        session_manager_service: SessionManagerService,
+        feature_namespace_service: FeatureNamespaceService,
+    ):
+        self.entity_service = entity_service
+        self.session_manager_service = session_manager_service
+        self.feature_namespace_service = feature_namespace_service
+
+    async def apply_comments(
+        self,
+        feature_store: FeatureStoreModel,
+        comments: Sequence[Union[TableComment, ColumnComment]],
+    ) -> None:
+        """
+        Add the provided table or column comments in the data warehouse
+
+        Parameters
+        ----------
+        feature_store: FeatureStoreModel
+            Feature store model
+        comments:  Sequence[Union[TableComment, ColumnComment]]
+            List of comments to be added
+        """
+        session = await self.session_manager_service.get_feature_store_session(feature_store)
+        for entry in comments:
+            if isinstance(entry, TableComment):
+                await session.comment_table(entry.table_name, entry.comment)
+            else:
+                await session.comment_column(entry.table_name, entry.column_name, entry.comment)
+
+    async def generate_table_comment(
+        self, feature_table_model: OfflineStoreFeatureTableModel
+    ) -> TableComment:
+        """
+        Generate comment for an offline feature table
+
+        Parameters
+        ----------
+        feature_table_model: OfflineStoreFeatureTableModel
+            Offline store feature table model
+
+        Returns
+        -------
+        TableComment
+        """
+        primary_entities = await self.entity_service.get_entities(
+            set(feature_table_model.primary_entity_ids)
+        )
+
+        def _format_entity(entity_model: EntityModel) -> str:
+            return f"{entity_model.name} (serving name: {entity_model.serving_names[0]})"
+
+        primary_entities_info = ", ".join([_format_entity(entity) for entity in primary_entities])
+        if feature_table_model.entity_lookup_info is not None:
+            parent_entity = await self.entity_service.get_document(
+                feature_table_model.entity_lookup_info.related_entity_id
+            )
+            sentences = [
+                f"This feature table is used to lookup the entity {_format_entity(parent_entity)}"
+                f" using the entity {primary_entities_info} based on their parent-child"
+                f" relationship"
+            ]
+        else:
+            sentences = [
+                f"This feature table consists of features for primary entity {primary_entities_info}"
+            ]
```

Comment:
Just to double check whether it is expected. If the table doesn't have primary entity, the `primary_entities_info` will be empty string.

Reply:
Good point, will update to handle the case where table doesn't have primary entity.